pkg/csi/cinder/openstack/openstack.go

```diff
@@ -197,24 +207,25 @@ func CreateOpenStackProvider() (IOpenStack, error) {
 	}
 
 	// Init OpenStack
-	OsInstance = &OpenStack{
+	OsInstances[cloudName] = &OpenStack{
 		compute:      computeclient,
 		blockstorage: blockstorageclient,
 		bsOpts:       cfg.BlockStorage,
 		epOpts:       epOpts,
 		metadataOpts: cfg.Metadata,
 	}
 
-	return OsInstance, nil
+	return OsInstances[cloudName], nil
 }
 
 // GetOpenStackProvider returns Openstack Instance
-func GetOpenStackProvider() (IOpenStack, error) {
-	if OsInstance != nil {
+func GetOpenStackProvider(cloudName string) (IOpenStack, error) {
+	OsInstance, OsInstanceDefined := OsInstances[cloudName]
+	if OsInstanceDefined {
 		return OsInstance, nil
 	}
 	var err error
```

Comment:
You can use := in the next line to skip this declaration.

Reply:
00bb738a26f48740410451f4e7e635643047995f